src/textual/widgets/_list_view.py

```diff
@@ -146,9 +146,11 @@ class Highlighted(Message, bubble=True):
         """Emitted when the highlighted item changes. Highlighted item is controlled using up/down keys.
 
         Attributes:
-            item (ListItem | None): The highlighted item, if there is one highlighted.
+            item: The highlighted item, if there is one highlighted.
         """
 
+        item: ListItem | None
+
```

Comment:
Does doing this hoist the type up in the the attributes in the class docstring? Testing stand-alone here I'm finding that this:

```python
class MyClass:
    """Some class.

    Attributes:
       some_attr: This is some instance attribute.
    """

    def __init__( self ):
        self.some_attr: float = 0.0
```

does the trick but this:

```python
class MyClass:
    """Some class.

    Attributes:
       some_attr: This is some instance attribute.
    """

    some_attr: float

    def __init__( self ):
        self.some_attr = 0.0
```

doesn't.

Reply:
Also, for what it's worth, this works too, but highlights the individual instance attributes rather than makes a table:

```python
class MyClass:
    """Some class."""

    some_attr: float

    def __init__( self ):
        self.some_attr = 0.0
        """This is some instance attribute."""
```

Reply:
You are right. The second option isn't working... Which is weird, because I tested it before going for that type of change. I must've done something wrong when testing.

In that case, I'd say it's open for debate what we go with:

 - Option A

```py
class MyClass:
    """This is my docstring:

    Attributes:
        my_attr (float): This is what my_attr does.
    """

    def __init__(self, *args, **kwargs):
        self.my_attr = ...
```

 - Option B

```py
class MyClass:
    """This is my docstring:

    Attributes:
        my_attr: This is what my_attr does.
    """

    def __init__(self, *args, **kwargs):
        self.my_attr: float = ...
```

I recommend option B because type checkers will detect if that becomes wrong, whereas option A can become a lie without us ever noticing.
@willmcgugan what have you say?

Reply:
I haven't been following. Suffice to say it is essential not to break type checkers. Over that we should do whatever works best in out docs.

If you and @davep agree a way forward I'll +1 it.

Reply:
I'd 100% go for option B -- it's a shame it doesn't infer the type from the value assigned in the same way that a type checker would; but explicit typing is always better than no typing. Count me in for this.